src/equations/ideal_glm_mhd_multiion_3d.jl

```diff
@@ -985,6 +1159,178 @@ function flux_ruedaramirez_etal(u_ll, u_rr, orientation::Integer,
+               (rho_e_rr - 0.5f0 * rho_rr * vel_norm_rr - 0.5f0 * mag_norm_rr -
+                0.5f0 * psi_rr^2)
+        beta_ll = 0.5f0 * rho_ll / p_ll
+        beta_rr = 0.5f0 * rho_rr / p_rr
+
+        # for convenience store vk_plus⋅B
+        vel_dot_mag_ll = vk1_plus_ll[k] * B1_ll + vk2_plus_ll[k] * B2_ll +
+                         vk3_plus_ll[k] * B3_ll
+        vel_dot_mag_rr = vk1_plus_rr[k] * B1_rr + vk2_plus_rr[k] * B2_rr +
+                         vk3_plus_rr[k] * B3_rr
+
+        # Compute the necessary mean values needed for either direction
+        rho_avg = 0.5f0 * (rho_ll + rho_rr)
+        rho_mean = ln_mean(rho_ll, rho_rr)
+        beta_mean = ln_mean(beta_ll, beta_rr)
+        beta_avg = 0.5f0 * (beta_ll + beta_rr)
+        p_mean = 0.5f0 * rho_avg / beta_avg
+        v1_avg = 0.5f0 * (v1_ll + v1_rr)
+        v2_avg = 0.5f0 * (v2_ll + v2_rr)
+        v3_avg = 0.5f0 * (v3_ll + v3_rr)
+        vel_norm_avg = 0.5f0 * (vel_norm_ll + vel_norm_rr)
+        vel_dot_mag_avg = 0.5f0 * (vel_dot_mag_ll + vel_dot_mag_rr)
+        vk1_plus_avg = 0.5f0 * (vk1_plus_ll[k] + vk1_plus_rr[k])
+        vk2_plus_avg = 0.5f0 * (vk2_plus_ll[k] + vk2_plus_rr[k])
+        vk3_plus_avg = 0.5f0 * (vk3_plus_ll[k] + vk3_plus_rr[k])
+        # v_minus
+        vk1_minus_ll = v1_plus_ll - vk1_plus_ll[k]
+        vk2_minus_ll = v2_plus_ll - vk2_plus_ll[k]
+        vk3_minus_ll = v3_plus_ll - vk3_plus_ll[k]
+        vk1_minus_rr = v1_plus_rr - vk1_plus_rr[k]
+        vk2_minus_rr = v2_plus_rr - vk2_plus_rr[k]
+        vk3_minus_rr = v3_plus_rr - vk3_plus_rr[k]
+        vk1_minus_avg = 0.5f0 * (vk1_minus_ll + vk1_minus_rr)
+        vk2_minus_avg = 0.5f0 * (vk2_minus_ll + vk2_minus_rr)
+        vk3_minus_avg = 0.5f0 * (vk3_minus_ll + vk3_minus_rr)
+        # Compute terms that depend on normal direction
+        v_dot_n_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2] +
+                     v3_ll * normal_direction[3]
+        v_dot_n_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2] +
+                     v3_rr * normal_direction[3]
+        vk_plus_mag_avg_dot_n = 0.5f0 * (normal_direction[1] *
+                                 (vk1_plus_ll[k] * mag_norm_ll +
+                                  vk1_plus_rr[k] * mag_norm_rr) +
+                                 normal_direction[2] *
+                                 (vk2_plus_ll[k] * mag_norm_ll +
+                                  vk2_plus_rr[k] * mag_norm_rr) +
+                                 normal_direction[3] *
+                                 (vk3_plus_ll[k] * mag_norm_ll +
+                                  vk3_plus_rr[k] * mag_norm_rr))
+        vk_plus_dot_n_avg = vk1_plus_avg * normal_direction[1] +
+                            vk2_plus_avg * normal_direction[2] +
+                            vk3_plus_avg * normal_direction[3]
+
+        # Fill the fluxes for the mass and momentum equations
+        f1 = rho_mean * 0.5 * (v_dot_n_ll + v_dot_n_rr)
```

Comment:
```suggestion
        f1 = rho_mean * 0.5f0 * (v_dot_n_ll + v_dot_n_rr)
```

Reply:
Good catch, @ranocha. Many thanks!
I wonder why the type test was passing! 🤔